sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -237,10 +237,15 @@ impl InputSelection {
         let remainder_address = self.get_remainder_address()?.map(|v| v.0);
 
         // Mana can potentially be added to an appropriate existing output instead of a new remainder output
-        let mana_remainder = selected_mana > required_mana
+        let mut mana_remainder = selected_mana > required_mana
             && remainder_address.map_or(true, |remainder_address| {
                 !self.output_for_added_mana_exists(&remainder_address)
             });
+        // If we are burning mana, we may not need a mana remainder
+        if self.burn.as_ref().map_or(false, |b| b.mana()) {
+            let initial_excess = self.initial_mana_excess()?;
+            mana_remainder &= selected_mana > required_mana + initial_excess;
```

Comment:
&= is bitwise but then it also works for booleans?

Reply:
booleans are bits...kiiiind of

but yeah it works the same